Assets/MRTK/SDK/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -44,33 +45,112 @@ public MigrationTool()
         /// Adds selectedObject to the list of objects to be migrated. Return false if the object is not of type GameObject, or SceneAsset.
         /// </summary>
         /// <param name="selectedObject">Unity Object of type GameObject or SceneAsset</param>
-        public bool TryAddObjectForMigration(Object selectedObject)
+        public bool TryAddObjectForMigration(Type type, Object selectedObject)
         {
+            if(type == null)
+            {
+                Debug.LogError("Migration type needs to be selected before migration.");
+                return false;
+            }
+
+            if(type != migrationHandlerInstanceType)
+            {
+                ClearMigrationList();
+                Debug.LogWarning("New migration type selected for migration. Clearing previous selection.");
+
+                if (!SetMigrationHandlerInstance(type))
+                {
+                    return false;
+                }
+            }
+
             if (!selectedObject)
             {
-                Debug.LogError("Selection is Empty. Could not add for migration.");
+                Debug.LogError("Selection is empty. Please select object for migration.");
                 return false;
             }
 
             if (selectedObject is GameObject || selectedObject is SceneAsset)
             {
-                if (!migrationObjects.Contains(selectedObject))
+                if (CheckIfCanMigrate(type, selectedObject) && !migrationObjects.Contains(selectedObject))
                 {
                     migrationObjects.Add(selectedObject);
-                    Debug.Log($"{selectedObject.name} object added for migration");
+                    return true;
+                }
+                else
+                {
+                    Debug.Log($"{selectedObject.name} does not support {type.Name} migration. Could not add object for migration");
+                    return false;
                 }
-                return true;
             }
             Debug.LogError("Object must be a GameObject, Prefab or SceneAsset. Could not add object for migration");
             return false;
         }
 
+        private bool CheckIfCanMigrate(Type type, Object selectedObject)
+        {
+            string objectPath = AssetDatabase.GetAssetPath(selectedObject);
+
+            // Object migrationObject is a scene game object
+            if (String.IsNullOrEmpty(objectPath) && selectedObject is GameObject)
+            {
+                var objectHierarchy = ((GameObject)selectedObject).GetComponentsInChildren<Transform>();
+                for (int i = 0; i < objectHierarchy.Length; i++)
+                {
+                    try
+                    {
+                        if (migrationHandlerInstance.CanMigrate(objectHierarchy[i].gameObject))
+                        {
+                            return true;
+                        }
+                    }
+                    catch (Exception e)
```

Comment:
Out of curiousity, what is the expected exception you'd get here? Just in general it's weird for a function to both return true/false and possibly throw.